skimage/data/generate_shapes.py

```diff
@@ -0,0 +1,328 @@
+
+    return triangle, label
+
+
+# Allows lookup by key as well as random selection.
+SHAPE_GENERATORS = dict(
+    rectangle=_generate_rectangle_mask,
+    circle=_generate_circle_mask,
+    triangle=_generate_triangle_mask)
+SHAPE_CHOICES = list(SHAPE_GENERATORS.values())
+
+
+def _generate_random_color(gray, min_pixel_intensity, random):
+    """Generates a random color array.
+
+    Parameters
+    ----------
+    gray : bool
+        If `True`, the color will be a scalar, else a 3-element array.
+    min_pixel_intensity : [0-255] int
+        The lower bound for the pixel values.
+    random : np.random.RandomState
+        The random state to use for random sampling.
+
+    Raises
+    ------
+    ValueError
+        When the min_pixel_intensity is not in the interval [0, 255].
+
+    Returns
+    -------
+    color : scalar or array
+        If gray is True, a single random scalar in the range of
+        [min_pixel_intensity, 255], else an array of three elements, each in
+        the range of [min_pixel_intensity, 255].
+
+    """
+    if not (0 <= min_pixel_intensity <= 255):
+        raise ValueError('Minimum intensity must be in interval [0, 255]')
+    if gray:
+        return random.randint(min_pixel_intensity, 256)
+    return random.randint(min_pixel_intensity, 256, size=3)
+
+
+def generate_shapes(image_shape,
+                    max_shapes,
+                    min_shapes=1,
+                    min_size=2,
+                    max_size=None,
+                    gray=False,
+                    shape=None,
+                    min_pixel_intensity=0,
+                    allow_overlap=False,
+                    number_of_attemps_to_fit_shape=100,
+                    random_seed=None):
+    """Generate an image with random shapes, labeled with bounding boxes.
+
+    The image is populated with random shapes with random sizes, random
+    locations, and random colors, with or without overlap.
+
+    Shapes have random (row, col) starting coordinates and random sizes bounded
+    by `min_size` and `max_size`. It can occur that a randomly generated shape
+    will not fit the image at all. In that case, the algorithm will try again
+    with new starting coordinates a certain number of times. However, it also
+    means that some shapes may be skipped altogether. In that case, this
+    function will generate fewer shapes than requested.
+
+    Parameters
+    ----------
+    image_shape : (int, int)
+        The number of rows and columns of the image to generate.
+    max_shapes : int
+        The maximum number of shapes to (attempt to) fit into the shape.
+    min_shapes : int
+        The minimum number of shapes to (attempt to) fit into the shape.
+    min_size : int
+        The minimum dimension of each shape to fit into the image.
+    max_size : int
+        The maximum dimension of each shape to fit into the image.
+    gray : bool
+        If `True`, generate 1-D monochrome images, else 3-D RGB images.
```

Comment:
I am not sure I understand what "3-D RGB images" are. If you simply mean 2D + color channels, maybe call it a "color image with multiple channels", since it is probably not limited to RGB but any colorspace?